### PR TITLE
For chinese aishell1 egs

### PR DESCRIPTION
**Pull request by zhaomingwork**
add aishell1 egs. you should set NUM_TEXT_TOKENS = 1280 in valle.py. Attached is the generated wav demo. 
[infer_demo.zip](https://github.com/lifeiteng/vall-e/files/11085143/infer_demo.zip)

### CODE REVIEW

File: egs/aishell1/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+
+## Inference
+
+```
+python3 bin/infer.py --output-dir infer/demos \
+    --model-name valle --norm-first true --add-prenet false \
+    --decoder-dim 1024 --nhead 16 --num-decoder-layers 12  \
+    --text-prompts "KNOT one point one five miles per hour." \
```
Comment (lifeiteng) [inline egs/aishell1/README.md:36]: 大家好非常高兴能认识大家

File: egs/aishell1/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    --audio-prompts ./prompts/8463_294825_000043_000000.wav \
```
Comment (lifeiteng) [inline egs/aishell1/README.md:37]: `./prompts/ch_24k.wav`

File: egs/aishell1/run.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  log "Stage 0: Download data"
+
+  # If you have pre-downloaded it to /path/to/aishell,
+  # you can create a symlink
+  #
+  #   ln -sfv /path/to/aishell $dl_dir/aishell
+  #
+  if [ ! -d $dl_dir/aishell/dev ]; then
+    # lhotse download aishell $dl_dir
+    lhotse download aishell   $dl_dir
```
Comment (lifeiteng) [inline egs/aishell1/run.sh:70]: format

File: egs/aishell1/run.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        --output-dir "${audio_feats_dir}"
+  fi
+  touch ${audio_feats_dir}/.aishell.tokenize.done
+fi
+
+if [ $stage -le 3 ] && [ $stop_stage -ge 3 ]; then
+  log "Stage 3: Prepare aishell train/dev/test"
+  if [ ! -e ${audio_feats_dir}/.aishell.train.done ]; then
+
+      # train
```
Comment (lifeiteng) [inline egs/aishell1/run.sh:104]: format

we should add `precommit for shell scripts`

File: egs/aishell1/run.sh [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    dtype="float16"
+  fi
+
+  python bin/trainer.py --manifest-dir ${audio_feats_dir} \
+    --text-tokens ${audio_feats_dir}/unique_text_tokens.k2symbols \
+    --max-duration ${max_duration} --filter-max-duration ${filter_max_duration} --dtype ${dtype} \
+    --model-name "${model_name}" --norm-first true --add-prenet false \
+    --decoder-dim ${decoder_dim} --nhead ${nhead} --num-decoder-layers ${num_decoder_layers} --prefix-mode ${prefix_mode} \
+    --accumulate-grad-steps ${accumulate_grad_steps} --base-lr ${base_lr} \
+    --num-epochs ${num_epochs} --start-epoch 1 --start-batch 0 --world-size 7\
```
Comment (lifeiteng) [inline egs/aishell1/run.sh:136]: add hparam `--world-size` for run.sh

File: egs/aishell1/shared/parse_options.sh [hunk 6]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
```
Comment (lifeiteng) [inline egs/aishell1/shared/parse_options.sh:2]: `ln -s ../libritts/shared/`

File: egs/aishell1/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# aishell1
+if you use aishell pypinyin_g2p Tokenizer you should set NUM_TEXT_TOKENS = 1280 in valle.py
+
+## Install deps
+```
+pip install librosa==0.8.1
```
Comment (lifeiteng) [inline egs/aishell1/README.md:6]: +pypinyin

**Review by lifeiteng — CHANGES_REQUESTED** [review]

File: egs/aishell1/run.sh [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+nj=16
+stage=2
+stop_stage=2
+
+# We assume dl_dir (download dir) contains the following
+# directories and files. If not, they will be downloaded
+# by this script automatically.
+#
+#  - $dl_dir/aishell
+#      You can download aishell from https://www.openslr.org/60/
```
Comment (lifeiteng) [inline egs/aishell1/run.sh:17]: /33

Comment (zhaomingwork) [reply egs/aishell1/run.sh]: OK. Commited all to one.

File: valle/tests/data/pypinyin_test.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            "e4",
+            "sh",
+            "i4",
+        ]
+        phonemes = text_tokenizer(input)
+        assert phonemes == output
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (lifeiteng) [inline valle/tests/data/pypinyin_test.py:51]: fine

**Review by lifeiteng — APPROVED** [review]

**Comment by sanqianyuejia** [comment]
> add aishell1 egs. you should set NUM_TEXT_TOKENS = 1280 in valle.py. Attached is the generated wav demo. [infer_demo.zip](https://github.com/lifeiteng/vall-e/files/11085143/infer_demo.zip)

Errors will occur due to the existence of english words in the corpus text.

**Comment by zhaomingwork** [comment]
> > add aishell1 egs. you should set NUM_TEXT_TOKENS = 1280 in valle.py. Attached is the generated wav demo. [infer_demo.zip](https://github.com/lifeiteng/vall-e/files/11085143/infer_demo.zip)
> 
> Errors will occur due to the existence of english words in the corpus text.

Had fixed a bug for SPACE problem, you can try again.  Or you can tell me the detail about the Error. Thx!

**Pull request merged**